balta/src/test/scala/za/co/absa/db/balta/classes/QueryResultRowIntegrationTests.scala

```diff
@@ -0,0 +1,221 @@
+    //second row
+    val secondRow = tableRows.tail.head
+    assert(secondRow.getBigDecimal(8).isEmpty)
+    assert(secondRow.getBigDecimal("bigdecimal_type").isEmpty)
+  }
+
+  test("getDate") {
+    //first row
+    val df = new SimpleDateFormat("yyyy-MM-dd");
+    val expected = new Date(df.parse("2022-08-09").getTime)
+    assert(tableRows.head.getDate(9).contains(expected))
+    assert(tableRows.head.getDate("date_type").contains(expected))
+    //second row
+    val secondRow = tableRows.tail.head
+    assert(secondRow.getDate(9).isEmpty)
+    assert(secondRow.getDate("date_type").isEmpty)
+  }
+
+  test("getTime") {
+    //first row
+    val df = new SimpleDateFormat("hh:mm:ss")
+    val expected = new Time(df.parse("10:12:15").getTime)
+    assert(tableRows.head.getTime(10).contains(expected))
+    assert(tableRows.head.getTime("time_type").contains(expected))
+    //second row
+    val secondRow = tableRows.tail.head
+    assert(secondRow.getTime(10).isEmpty)
+    assert(secondRow.getTime("time_type").isEmpty)
+  }
+
+
+  test("getLocalDateTime") {
+    //first row
+    val timestampString = "2020-06-02T01:00:00"
+    val expectedLocalDateTime = LocalDateTime.parse(s"$timestampString")
+
+    assert(tableRows.head.getLocalDateTime(11).contains(expectedLocalDateTime))
+    assert(tableRows.head.getLocalDateTime("timestamp_type").contains(expectedLocalDateTime))
+
+    //second row
+    val secondRow = tableRows.tail.head
+    assert(secondRow.getLocalDateTime(11).isEmpty)
+    assert(secondRow.getLocalDateTime("timestamp_type").isEmpty)
+  }
+
+  test("getOffsetDateTime and getInstant") {
+    //first row
+    val expectedOffsetDateTime = OffsetDateTime.parse("2021-04-03T11:00:00+01:00")
+    val expectedInstant = Instant.parse("2021-04-03T10:00:00.00Z")
+
+    val resultOfColumn = tableRows.head.getOffsetDateTime(12).get
+    val resultOfColumnLabel = tableRows.head.getOffsetDateTime("timestamptz_type").get
+    assert(resultOfColumn.isEqual(expectedOffsetDateTime))
+    assert(resultOfColumnLabel.isEqual(expectedOffsetDateTime))
+    assert(tableRows.head.getInstant(12).contains(expectedInstant))
+    assert(tableRows.head.getInstant("timestamptz_type").contains(expectedInstant))
+    //second row
+    val secondRow = tableRows.tail.head
+    assert(secondRow.getOffsetDateTime(12).isEmpty)
+    assert(secondRow.getOffsetDateTime("timestamptz_type").isEmpty)
+    assert(secondRow.getInstant(12).isEmpty)
+    assert(secondRow.getInstant("timestamptz_type").isEmpty)
+  }
+
+  test("getUUID") {
+    //first row
+    val expected = UUID.fromString("090416f8-7da0-4598-844b-63659334e5b6")
+    assert(tableRows.head.getUUID(13).contains(expected))
+    assert(tableRows.head.getUUID("uuid_type").contains(expected))
+    //second row
+    val secondRow = tableRows.tail.head
+    assert(secondRow.getUUID(13).isEmpty)
+    assert(secondRow.getUUID("uuid_type").isEmpty)
+  }
+
+  test("getVector[Int]") {
```

Comment:
We are testing a `getArray` method using a `Vector`. The name should be related to the tested method, not the used solution.